opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+/**
+ * Tracer is the entry point API between instrumentation code and the tracing implementation.
+ */
+public interface Tracer {
```

Comment:
In effective java, which a lot modern java takes from, composition is
favored over inheritance. I was already a bit troubled, writing this, as
another inheritance wants an one object to also be a codec.

I think if we weren't explicit about requiring inheritance, we can write
and wire cleaner components in java.